storage/api/src/main/java/feast/storage/api/retrieval/OnlineRetriever.java

```diff
@@ -0,0 +1,39 @@
+package feast.storage.api.retrieval;
+
+import feast.serving.ServingAPIProto.GetOnlineFeaturesRequest;
+import feast.serving.ServingAPIProto.GetOnlineFeaturesResponse;
+import feast.types.FeatureRowProto;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Interface for implementing user defined retrieval functionality from
+ * Online stores.
+ */
+public interface OnlineRetriever<T> extends Serializable {
+
+    /**
+     * Get all values corresponding to the request.
+     *
+     * @param request Online features request containing the list of features and list of entity rows
+     * @return list of {@link T}
+     */
+    List<T> getFeatures(GetOnlineFeaturesRequest request);
```

Comment:
In the case of a JDBC (lets say SQLite store), which interfaces could possibly be used together here? For example could you have have OnlineRetriever, BatchRetriever, DeadletterSink all being implemented by a single class?

If that is the case it may be a little clearer to use different method names instead of overloading, but I dont feel super strongly about it.

Reply:
I'd highly recommend that implementations be separate, but I guess it's a possibility that people just want to implement it all in a single class :/. Let me change it...

Reply:
I think it would be quite powerful if we could support it. Specifically if we could configure a store once (perhaps in `application.yaml`), enable its different modes, and then pass it around throughout the codebase.